taipy/core/_orchestrator/_orchestrator.py

```diff
@@ -223,18 +227,25 @@ def _unlock_edit_on_jobs_outputs(jobs: Union[Job, List[Job], Set[Job]]):
     @classmethod
     def _on_status_change(cls, job: Job):
         if job.is_completed() or job.is_skipped():
+            cls.__logger.debug(f"{job.id} has been completed or skipped. Unblocking jobs.")
             cls.__unblock_jobs()
         elif job.is_failed():
             cls._fail_subsequent_jobs(job)
 
     @classmethod
     def __unblock_jobs(cls):
-        for job in cls.blocked_jobs:
-            if not cls._is_blocked(job):
-                with cls.lock:
+        with cls.lock:
```

Comment:
The lock is global now to ensure all the transactionality of the unblock operation.